zerver/lib/test_helpers.py

```diff
@@ -314,44 +314,31 @@ class DummyHandler(AsyncDjangoHandler):
     def __init__(self) -> None:
         allocate_handler_id(self)
 
+class HostRequestMock:
+    """A mock request object where get_host() works.  Useful for testing
+    routes that use Zulip's subdomains feature"""
 
-class POSTRequestMock:
-    method = "POST"
-
-    def __init__(self, post_data: Dict[str, Any], user_profile: Optional[UserProfile]) -> None:
+    def __init__(self, post_data: Dict[str, Any]={}, user_profile: Optional[UserProfile]=None, host: str=settings.EXTERNAL_HOST) -> None:
+        self.host = host
         self.GET: Dict[str, Any] = {}
+        self.method = ''
 
         # Convert any integer parameters passed into strings, even
         # though of course the HTTP API would do so.  Ideally, we'd
         # get rid of this abstraction entirely and just use the HTTP
-        # API directly, but while it exists, we need this code.
-        self.POST: Dict[str, str] = {}
+        # API directly, but while it exists, we need this code
+        self.POST: Dict[str, Any] = {}
         for key in post_data:
             self.POST[key] = str(post_data[key])
+            self.method = "POST"
 
-        self.user = user_profile
         self._tornado_handler = DummyHandler()
         self._log_data: Dict[str, Any] = {}
-        self.META = {"PATH_INFO": "test"}
-        self.path = ""
-
-
-class HostRequestMock:
-    """A mock request object where get_host() works.  Useful for testing
-    routes that use Zulip's subdomains feature"""
-
-    def __init__(
-        self, user_profile: Optional[UserProfile] = None, host: str = settings.EXTERNAL_HOST
-    ) -> None:
-        self.host = host
-        self.GET: Dict[str, Any] = {}
-        self.POST: Dict[str, Any] = {}
-        self.META = {"PATH_INFO": "test"}
-        self.path = ""
+        self.META = {'PATH_INFO': 'test'}
+        self.path = ''
         self.user = user_profile
-        self.method = ""
-        self.body = ""
-        self.content_type = ""
+        self.body = ''
+        self.content_type = ''
```

Comment:
Can you write a bit in the commit message about the strategy here for what parts are taken from where?

One thing I notice is that you initialize `_tornado_handler` in HostRequestMock; I'd prefer not to do that, as it's only correct for Tornado requests, which are very rare.  I think what I'd like to see is a series of commits here that migrate individual features, e.g. a first one that adds a `post_data=` optional parameter to HostRequestMock, and migrates a few callers that will work with just that change, and later one that allows passing `attach_tornado_handler=True` to add the `self._tornado_handler` hack, which I imagine would only be called in a small number of places.


Does that make sense?  

Check out the Zulip commit guidelines for more details: https://zulip.readthedocs.io/en/latest/contributing/version-control.html

Reply:
Of course! Thanks for the feedback!

The strategy was to directly transplant everything from POSTRequestMock into HostRequestMock, since nothing specific was requested in the initial comments of the issue. This is why I initialized _tornado_handler. Are there variables/features of the POSTRequestMock I shouldn't worry about integrating into HostRequestMock?

To make sure I understand correctly:  
1. I should revert my changes
2. Make a commit for each feature migration and the associated callers' migration

Reply:
Also, I fixed the lint issues from before and all tests are passing now.